src/shared/bg_pmove.cpp

```diff
@@ -4755,8 +4711,23 @@ void PmoveSingle( pmove_t *pmove )
 	}
 	else if ( pml.walking )
 	{
-		if ( BG_ClassHasAbility( pm->ps->stats[ STAT_CLASS ], SCA_WALLCLIMBER ) &&
-		     ( pm->ps->stats[ STAT_STATE ] & SS_WALLCLIMBING ) )
+		if ( pm->waterlevel > 2 && DotProduct( pml.forward, pml.groundTrace.plane.normal ) > 0 )
```

Comment:
This one `PM_ClimbMove PM_WalkMove dedup (to PmoveSingle)` is not correct because there were early returns from PM_WalkMove, but now the rest of the function will always execute.

Reply:
Good catch, needs an additional "else" block there